smithy-typescript-ssdk-libs/server-common/src/validation/validators.ts

```diff
@@ -13,16 +13,57 @@
  *  permissions and limitations under the License.
  */
 
-import { EnumValidationFailure, LengthValidationFailure, PatternValidationFailure, UniqueItemsValidationFailure } from ".";
+import {
+    EnumValidationFailure,
+    LengthValidationFailure,
+    PatternValidationFailure,
+    RangeValidationFailure,
+    RequiredValidationFailure,
+    UniqueItemsValidationFailure,
+    ValidationFailure
+} from ".";
+
+export class CompositeValidator<T> {
+    private readonly validators: Validator<T, any>[];
+
+    constructor(validators: Validator<T, any>[]) {
+        this.validators = validators;
+    }
+
+    validate(input: T | undefined | null, memberName: string): ValidationFailure[] {
+        let retVal: ValidationFailure[] = [];
+        for (let v of this.validators) {
+            let failure = v.validate(input, memberName);
+            if (failure) {
+                retVal.push(failure);
+            }
+        }
+        return retVal;
+    }
+}
+
+export class NullValidator {
```

Comment:
`NoOpValidator`? I say this because validating nullability is feasibly something this could be doing

Reply:
Isn't that what RequiredValidator does? This is the null object pattern, hence the name, but I have no qualms renaming it to NoOpValidator.